apps/frontend/static_src/js/main.js

```diff
@@ -1 +1,67 @@
 import { Dropdown } from 'bootstrap';
+
+const search_input = document.getElementById("search_input")
+const searchIconButton = document.getElementById("searchIconButton")
+
+const onSearchInputChange = async (event) => {
+    const query = event.target.value
+    try {
+        const res = await fetch(window.location.origin + '/search_json/?query=' + query)
+        const data = await res.json()
+        console.log(data)
+        injectResultsInHTML(data)
+    } catch(err) {
+        console.log(err)
+        return {"error": err}
+    }
+}
+search_input.addEventListener("keyup", onSearchInputChange)
+
+
+const removeExistingChildren = (parent) => {
+    const children = [...parent.children]
+    children.map((child) => {
+        child.remove()
+    })
```

Comment:
The `map()` method is used to transform a list of objects into another list of objects, so I don't think it's appropriate here. A simple for loop could work?

```suggestion
    for (child of parent.children) {
        child.remove();
    }
```